tests/pass/stacked-borrows/stack-printing.rs

```diff
@@ -1,4 +1,4 @@
-//@compile-flags: -Zmiri-permissive-provenance
+//@compile-flags: -Zmiri-permissive-provenance -Zmiri-provenance-gc=0
```

Comment:
Previously, the GC didn't run on the relevant stacks during the `-Zmiri-provenance-gc=1` test. Now it does. So adding this flag gives us consistent behavior.

Reply:
And it's a borrow-stack-printing test, those generally shouldn't GC as that changes the stacks. But please add a comment explaining that.